stdlib.sh

```diff
@@ -351,12 +351,12 @@ path_add() {
   local path i var_name="$1"
   # split existing paths into an array
   declare -a path_array
-  IFS=: read -ra path_array <<<"${!1}"
+  IFS=: read -ra path_array <<<"${!1-}"
   shift
 
   # prepend the passed paths in the right order
   for ((i = $#; i > 0; i--)); do
-    path_array=("$(expand_path "${!i}")" "${path_array[@]}")
+    path_array=("$(expand_path "${!i}")" ${path_array[@]+"${path_array[@]}"})
```

Comment:
I'm not sure about this solution. It seems like it's for bash 3.0 who is on macOS.

Reply:
Did you take a look at the SO post in the OP? That seems to suggest that this works on Bash 3, 4, 5. I have only tested with Bash 4 though, and that seems to work fine.

Would you rather have me revert this change and just disable `set -u` for the tests that I added?

Reply:
> Would you rather have me revert this change and just disable set -u for the tests that I added?

I did this for now.

Reply:
It looks like https://github.com/direnv/direnv/issues/675 is having a similar issue.